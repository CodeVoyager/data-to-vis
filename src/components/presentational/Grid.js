import React from 'react';
import {
    Link
} from 'react-router-dom';

const Grid = () => {
    return (
        <div className="grid">
            <Link className="item" component="div" to="/bpi">
                BPI
            </Link>
        </div>
    );
};

export default Grid;