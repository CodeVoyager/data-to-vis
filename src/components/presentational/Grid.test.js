import React from 'react';
import ReactDOM from 'react-dom';
import Grid from './Grid';
import renderer from 'react-test-renderer';
import {
    MemoryRouter as Router
} from 'react-router-dom';

describe('Grid component', function () {
    const ENTRIES = [ '/', '/bpi'];

    it('renders without crashing', function () {
        const div = document.createElement('div');
        ReactDOM.render(
            <Router>
                <Grid/>
            </Router>
        , div);
    });

    it('renders correctly', function () {
        const tree1 = renderer.create(
            <Router>
                <Grid/>
            </Router>
        ).toJSON();
        expect(tree1).toMatchSnapshot();

        const tree2 = renderer.create(
            <Router initialEntries={ENTRIES} initialIndex={0}>
                <Grid/>
            </Router>
        ).toJSON();
        expect(tree2).toMatchSnapshot();

        const tree3 = renderer.create(
            <Router initialEntries={ENTRIES} initialIndex={1}>
                <Grid/>
            </Router>
        ).toJSON();
        expect(tree3).toMatchSnapshot();
    });
});