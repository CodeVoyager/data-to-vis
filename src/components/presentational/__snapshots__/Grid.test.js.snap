// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Grid component renders correctly 1`] = `
<div
  className="grid"
>
  <a
    className="item"
    component="div"
    href="/bpi"
    onClick={[Function]}
  >
    BPI
  </a>
</div>
`;

exports[`Grid component renders correctly 2`] = `
<div
  className="grid"
>
  <a
    className="item"
    component="div"
    href="/bpi"
    onClick={[Function]}
  >
    BPI
  </a>
</div>
`;

exports[`Grid component renders correctly 3`] = `
<div
  className="grid"
>
  <a
    className="item"
    component="div"
    href="/bpi"
    onClick={[Function]}
  >
    BPI
  </a>
</div>
`;
